weave-js/src/components/Button/Button.tsx

```diff
@@ -82,70 +88,85 @@ export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-          {children}
-          {endIcon ? <Icon name={endIcon} /> : null}
-        </button>
-      </Tailwind>
+      <button
+        ref={ref}
+        type="button"
+        className={twMerge(
+          classNames(
+            'night-aware',
+            "inline-flex items-center justify-center whitespace-nowrap rounded border-none font-['Source_Sans_Pro'] font-semibold",
+            'disabled:pointer-events-none disabled:opacity-35',
+            'focus-visible:outline focus-visible:outline-[2px] focus-visible:outline-teal-500',
+            {
+              // small
+              'gap-6 px-6 py-3 text-sm leading-[18px]': isSmall,
+              '[&_svg]:h-16 [&_svg]:w-16': isSmall,
+              'h-24 w-24 p-0': isSmall && isIconOnly,
+
+              // medium
+              'gap-10 px-10 py-4 text-base': isMedium,
+              '[&_svg]:h-18 [&_svg]:w-18': isMedium,
+              'h-32 w-32 p-0': isMedium && isIconOnly,
+
+              // large
+              'gap-8 px-12 py-8 text-base': isLarge,
+              'h-40 w-40 p-0': isLarge && isIconOnly,
+
+              // primary
+              'bg-teal-500 text-white hover:bg-teal-450': isPrimary,
+              'bg-teal-450': isPrimary && active,
+
+              // secondary & ghost
+              'bg-oblivion/[0.05] dark:bg-moonbeam/[0.05]': isSecondary,
+              'text-moon-800 dark:text-moon-200': isSecondary || isGhost,
+              'hover:bg-teal-300/[0.48] hover:text-teal-600 dark:hover:bg-teal-700/[0.48] dark:hover:text-teal-400':
+                isSecondary || isGhost,
+              'bg-teal-300/[0.48] text-teal-600 dark:bg-teal-700/[0.48] dark:text-teal-400':
+                (isSecondary || isGhost) && active,
+
+              // quiet
+              'text-moon-500': isQuiet,
+              'bg-oblivion/[0.05] text-moon-800 dark:bg-moonbeam/[0.05] dark:text-moon-200':
+                isQuiet && active,
+              'hover:text-moon-800 dark:hover:text-moon-200': isQuiet,
+              'hover:bg-oblivion/[0.05] dark:hover:bg-moonbeam/[0.05]': isQuiet,
+
+              // destructive
+              'bg-red-500 text-white hover:bg-red-450': isDestructive,
+              'bg-red-450': isDestructive && active,
+            },
+            className
+          )
+        )}
+        {...htmlAttributes}>
+        {startIcon ? <Icon name={startIcon} /> : null}
+        {children}
+        {endIcon ? <Icon name={endIcon} /> : null}
+      </button>
     );
 
+    const [isTooltipOpen, setIsTooltipOpen] = useState(false);
     if (tooltip) {
-      // span is needed so tooltip works on disabled buttons
-      return <Tooltip content={tooltip} trigger={<span>{button}</span>} />;
+      return (
+        <Tailwind style={wrapperStyles}>
+          <Tooltip.Provider>
+            <Tooltip.Root open={isTooltipOpen} onOpenChange={setIsTooltipOpen}>
+              <Tooltip.Trigger className="cursor-pointer" asChild>
```

Comment:
verified disabled state as well -- cursor will not be pointer if button is disabled